Makefile

```diff
@@ -308,7 +308,7 @@ test: export THANOS_TEST_ALERTMANAGER_PATH= $(ALERTMANAGER)
 test: check-git install-tool-deps
 	@echo ">> install thanos GOOPTS=${GOOPTS}"
 	@echo ">> running unit tests (without /test/e2e). Do export THANOS_TEST_OBJSTORE_SKIP=GCS,S3,AZURE,SWIFT,COS,ALIYUNOSS,BOS,OCI,OBS if you want to skip e2e tests against all real store buckets. Current value: ${THANOS_TEST_OBJSTORE_SKIP}"
-	@go test -race -timeout 15m $(shell go list ./... | grep -v /vendor/ | grep -v /test/e2e);
+	@go test -race -timeout 25m $(shell go list ./... | grep -v /vendor/ | grep -v /test/e2e);
```

Comment:
The downside of patching the upstream Makefile is that we may face conflicts in the future preventing the syncbot to do its job. But to unblock the CI, I guess we have no other choice (the alternative would be to hard-code the go test command in the ci-operator config but it has other drawbacks like not picking up updates from upstream tooling).
As a follow-up task, we could ask Thanos upstream to add an env variable to customize the go test timeout.

Reply:
Yeah, I do totally agree on that. Actually I was preparing a PR for that, let's see if it gets accepted and we'll backport later.

Reply:
Increasing the timeout will unblock us for sure, but I'm afraid doing will hide the real issue, I looked at the test durations of `github.com/thanos-io/thanos/pkg/store` which is timing out now, and they're all below `700s`

Reply:
And that's why we were having issues as the prow cluster may be overloaded, as it happened with `build04` and went over 900s

Reply:
For the record, running it locally also is below 700s, so that's why I'm blaming CI over this ;)

Reply:
i.e. https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_thanos/144/pull-ci-openshift-thanos-master-test-local/1767496278268186624/artifacts/test/build-log.txt 

```
ok  	github.com/thanos-io/thanos/pkg/store	984.344s
```